docs/src/main/asciidoc/extension-codestart.adoc

```diff
@@ -0,0 +1,259 @@
+
+The codestart should not have any business code, instead, it should contain some stub data/hello world to compile. The idea is to bring code that is the starting point to everyone using the extension.
+
+== Writing an Extension Codestart in Quarkus Core
+
+- Duplicate an existing https://github.com/quarkusio/quarkus/tree/main/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts[Quarkus core extension codestarts, window="_blank"]. `resteasy-qute-codestart` is a good base when the code will expose a web resource, in other cases, `config-yaml-codestart` is a better base. More info on the <<directory-structure>>.
+
+- Edit the <<codestart-yml>>:
+
+- Create the extension binding in the extension metadata (https://github.com/quarkusio/quarkus/blob/main/extensions/config-yaml/runtime/src/main/resources/META-INF/quarkus-extension.yaml#L12-L17[example, window="_blank"]). *Thanks to this, the codestart is added when the user selects the extension*
+
+- Add the readme <<readme-md>> section template.
+
+- Add the code in the language folder (it is recommended to at least provide java and kotlin). *You have to use org.acme as package name: <<org-acme-package>>*. It is possible to use <<qute-templates>> if needed.
+
+- Optionally, Add the `index.html` section template (<<index-html>>).
+
+- Optionally, add some resources (`./base` directory if they are non language specific)
+
+- Optionally, add the <<app-config>>.
+
+- Create an <<integration-test>> *You can use it to get your codestart generated as a project*.
+
+You can also use it for dev using the `buildAllProjectsForLocalUse` (It's disabled by default because already covered by `QuarkusCodestartBuildIT`).
+
+== Writing an Extension Codestart in the Quarkiverse or standalone
+
+For non-core extensions, the codestart is located in the runtime module (with special instruction in the pom.xml to generate a separated codestart artifact). https://github.com/ia3andy/aloha-code/[Here, window="_blank"] is an example extension with a codestart & its tests
+
+
+== Specific topics
+
+[#org-acme-package]
+=== Dynamic package name generation from org.acme
+
+
+You have to use `org.acme` as package name in your extension codestart sources. In the generated project, the user specified package will be used (and auto-replace `org.acme`).
+
+In all the source files (.java, .kt, .scala), it will be auto-replaced, you don't have to care. The package directory also gets auto-replace. If for some reason, another type of file needs the user package name, then you should use a <<qute-templates>> for it and `{project.package-name}` data placeholder (https://github.com/quarkusio/quarkus/blob/main/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/grpc-codestart/base/src/main/proto/hello.tpl.qute.proto#L4[find an example in the grpc proto file, window="_blank"]).
```

Comment:
```suggestion
It will be auto-replaced in all the source files (.java, .kt, .scala). The package directory will also be automatically adjusted. If for some reason, another type of file needs the user package name then you should use a <<qute-templates>> for it and `{project.package-name}` data placeholder (https://github.com/quarkusio/quarkus/blob/main/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/grpc-codestart/base/src/main/proto/hello.tpl.qute.proto#L4[find an example in the grpc proto file, window="_blank"]).
```